src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -115,14 +116,14 @@
     private static final Logger log = LoggerFactory.getLogger(MtAmazonDynamoDbBySharedTable.class);
 
     private final String name;
-
+    private final Optional<String> topLevelContext;
```

Comment:
What is the purpose of this context?

Reply:
This is so that we know whether we're allowed to create/delete/update a multitenant table or to do a cross-tenant scan on a multitenant table. For example this would be the Core cloud context.

Reply:
Is this different from "no context" (as used in master for cross-tenant scans or change streaming)?

Reply:
Yes this is different from "no context". It seems like when for example the Core cloud does DDL on a virtual multitenant table, or does a cross-tenant scan on a virtual mt table, there would still be a context.
But maybe we shouldn't make the distinction -- that would make things simpler.